### PR TITLE
feat: ✨ initial XCM implementation + xcm-simulator setup and tests

### PR DESCRIPTION
**Pull request by TDemeco**
This PR:
- Configures the StorageHub runtime with all the required XCM parameteres for both  `pallet_xcm` and `xcm_executor`, according to the required functionality that it has to provide.
- Sets up a whole XCM testing suite using xcm-simulator, complete with a relay chain, a sibling system chain and a sibling normal parachain.
- Provides tests for DOT transfering between StorageHub and other chains.
- Provides tests for pallet calls from other parachains into StorageHub.

TODOs:
- [x] Write more tests testing more StorageHub functionality over XCM, the skeleton is finally there so it will be a long task but not difficult
- [x] ~Discuss allowing having the relay chain as an allowed reserve for DOT. This would mean any parachain will be able to transfer DOT to us without having to go to the relay chain or a system chain first. This complicates the system a bit (we would have a sovereign account on the relay chain while being a trusted teleporter) and might not be needed considering AssetHub is also a system chain (so other parachains can go through it to reach us) but it improves UX for other parachain teams with not a lot of costs.~ -> Left for another PR in the future

### CODE REVIEW

File: xcm-simulator/src/parachain/mod.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    pub const ReservedXcmpWeight: Weight = Weight::from_parts(WEIGHT_REF_TIME_PER_SECOND.saturating_div(4), 0);
+    pub const ReservedDmpWeight: Weight = Weight::from_parts(WEIGHT_REF_TIME_PER_SECOND.saturating_div(4), 0);
+}
+
+impl mock_message_queue::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type XcmExecutor = XcmExecutor<XcmConfig>;
+}
+
+pub type LocalOriginToLocation =
```
[the remark this answers is not in the document]
Comment (TDemeco) [reply xcm-simulator/src/parachain/mod.rs:129]: Keep in mind this type is in the `parachain` runtime, which simulates another parachain different from StorageHub that might have different type assumptions

[the remark this answers is not in the document]
Comment (snowmead) [reply xcm-simulator/src/parachain/mod.rs:129]: Ah I see. Disregard what I said.

File: pallets/providers/Cargo.toml [hunk 2]
```diff
@@ -64,6 +64,7 @@ std = [
 	"sp-core/std",
 	"sp-io/std",
 	"sp-runtime/std",
+	"pallet-balances/std",
```
Comment (ffarall) [inline pallets/providers/Cargo.toml:67]: Move up in alphabetical order

File: runtime/src/configs/mod.rs [hunk 3]
```diff
@@ -26,6 +26,7 @@
 mod xcm_config;
 
 // Substrate and Polkadot dependencies
+use crate::PolkadotXcm;
```
Comment (ffarall) [inline runtime/src/configs/mod.rs:29]: Move down to `// Local module imports`

**Review by ffarall — APPROVED** [review]
Remarkable work with the setup for testing XCM!

**Pull request merged**